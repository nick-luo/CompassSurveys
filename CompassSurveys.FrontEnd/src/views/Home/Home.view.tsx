import React, { useState, useEffect } from 'react';
import { useHistory } from 'react-router-dom';
import { Container, Button, Row, Spinner } from 'react-bootstrap';
import { getSurveys } from '../../utils/api/SurveyApi';
import { Survey } from '../../models';

export const HomeView: React.FC = () => {
  const [surveys, setSurveys] = useState([] as Survey[]);
  const [isLoading, setLoading] = useState(false);
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      setLoading(true);
      const result = await getSurveys();
      setSurveys(result);
      setLoading(false);
    };

    try {
      fetchData();
    } catch (err) {
      console.log(err);
    }
  }, []);

  const handleClick = (surveyId: string) => {
    history.push('/survey/' + surveyId);
  };

  return (
    <Container>
      <h1>Compass Surveys</h1>
      {isLoading
        ? <Row className="justify-content-md-center">
            <Spinner animation="border" />
          </Row>
        : surveys.map((value, index) => {
          return (
            <Button
              key={`${index}`}
              value={value.id}
              variant="primary"
              onClick={e => handleClick((e.target as HTMLInputElement).value)}
              block
              data-testid={`button-${value.id}`}
            >
              {value.name}
            </Button>
          );
        })
      }
    </Container>
  );
};
