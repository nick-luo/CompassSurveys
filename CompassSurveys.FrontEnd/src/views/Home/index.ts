export { HomeView } from "./Home.view";
