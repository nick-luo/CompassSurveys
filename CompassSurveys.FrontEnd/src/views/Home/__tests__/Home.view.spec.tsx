import React from 'react';
import { cleanup, render, RenderResult } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { HomeView } from '../Home.view';

describe('HomeView', () => {
  const renderComponent = (): RenderResult => {
    return render(
      <MemoryRouter>
        <HomeView />
      </MemoryRouter>,
    );
  };

  afterEach(cleanup);

  it('should render without crashing', () => {
    // act
    const renderedComponent = renderComponent();

    // assert
    expect(renderedComponent.container.firstChild).toMatchSnapshot();
  });
});
