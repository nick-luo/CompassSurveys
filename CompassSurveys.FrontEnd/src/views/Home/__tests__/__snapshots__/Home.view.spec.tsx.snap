// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView should render without crashing 1`] = `
<div
  class="container"
>
  <h1>
    Compass Surveys
  </h1>
  <div
    class="justify-content-md-center row"
  >
    <div
      class="spinner-border"
    />
  </div>
</div>
`;
